include/seqan3/range/container/aligned_sequence_adaptor_constant_access.hpp

```diff
@@ -0,0 +1,688 @@
+    }
+    //!\}
+
+    /*!\name Projection between gap and alphabet space.
+     * \{
+    */
+    //!\brief Return gap-free sequence.
+    std::vector<value_type> get_underlying_sequence() const
+    {
+        return sequence;
+    }
+
+   /*!\brief Insert gap 'size' times at given index.
+   *
+   * Return false if given index exceeds sequence length minus one.
+   */
+   bool insert_gap(size_type pos, size_type size=1)
+   {
+        if (pos > this->size())
+            return false;
+        if (pos == this->size() && size == 1)
+            push_back(gap::GAP);
+        else
+            insert(pos, size, gap::GAP);
+        return true;
+    }
+
+    /*! Remove gap at position pos of length gap_len. Return false when
+    *
+    * Return false if index exceeds the sequence length or the symbol is not a gap.
+    */
+    bool remove_gap(size_type pos, size_type len)
+    {
+        if (pos >= size() || !gap_vector[pos])
+            return false;
+        gap_vector[pos] = 1;
+        return true;
+    }
+
+    /*\brief Map a compressed representation index to the aligned sequence index.
+    *
+    * Note that a i-th 0 corresponds to the i-th position in the compressed sequence.
+    * We therefore can directly use the select support structure initialized to
+    * count 0s (and not 1s like for the rank) to map to the aligned sequence space.
+    * E.g. '--TA-TA--' with input pos = 2 returns select<0>(2) = 5.
+    */
+    size_type map_to_aligned_position(size_type const pos) const
+    {
+        assert(pos < gap_vector.size());
+        return ss(pos + 1);
+    }
+
+    /*!\brief Map from gapped sequence position to index of compressed
+    * sequence representation.
+    *
+    * We use the rank support structure indexed for counting 1s and substract
+    * the number gaps in [0; position_gap] from the input position.
+    * E.g.           aligned sequence  | - A - - T
+    *                 position_gapped  | 0 1 2 3 4
+    *      map_to_underlying_position  |-1 0 0 0 1
+    * Note that a gap is mapped to the same position than the next preceeding
+    * non-gap symbol.
+    */
+    difference_type map_to_underlying_position(size_type const position_gapped) const
+    {
+        return static_cast<difference_type>(position_gapped) -
+        static_cast<difference_type>(rs.rank(std::min<size_type>(position_gapped+1, this->size())));
+    }
+    //!\}
+
+    /*!\name Random access sequence concept support.
+     * \{
+    */
+    //!\brief Return reference to aligned sequence for given index.
+    constexpr reference operator[](size_type const n) const // const noexcept(noexcept((*host)[pos+n]))
+    {
+        assert(n < size());
+        if (!gap_vector[n]){
+            size_type const pos = map_to_underlying_position(n);
+            return value_type(sequence[pos]);
+        }
+        return gap_symbol;
+    }
+
+    //!\brief Return reference to aligned sequence for given index.
+    value_type at(size_type const idx) const
```

Comment:
should also return reference type / const reference (even if this is defined as the same at the beginning).